src/argilla_server/schemas/v1/records.py

```diff
@@ -104,7 +104,7 @@ def check_user_id_is_unique(cls, values: Optional[List[UserResponseCreate]]) ->
 
         return values
 
-    @validator("metadata", pre=True)
+    @validator("metadata")
```

Comment:
Why removing the `pre=True`?

Reply:
With `pre=True` the validator will be executed before type-checking validation, doing the validator fails (since expects a dictionary). 

I've removed the `pre` setup and added some missing tests for this.